joomla/joomla-cms
# [4.x] Treeselect fixes

Pull Request for Issue #41309

### Summary of Changes

1. Fix the treeselect dropdowns not working
2. Fix the check/uncheck buttons not toggling checkboxes that had been manually changed (backported from https://github.com/joomla/joomla-cms/pull/42776)
3. Fix select/deselect changing the state of the current menu (backported from https://github.com/joomla/joomla-cms/pull/42776)

### Testing Instructions

1. Go to edit any module in the Module Manager
2. Navigate to the Menu Assignment tab
3. Change the assignment to "Only on selected pages"

**Test 1:**
Ensure the dropdown menu opens as shown below:
![image](https://github.com/joomla/joomla-cms/assets/2019801/b473aabc-35c6-4d60-8700-2f3cb80b4daa)

**Test 2:**
1. Open the dropdown menu and click "select" or "unselect"
All checkboxes (including the one parallel to the dropdown toggle) should be toggled

**Test 3:**
1. Check (or uncheck) any checkbox
2. Click the buttons shown above: 
![image](https://github.com/joomla/joomla-cms/assets/2019801/de599d4e-c34f-4f42-a083-13018b869ba5)

All checkboxes should be toggled.

@LadySolveig @Quy
